src/Libraries/Microsoft.AspNetCore.Diagnostics.Middleware/Logging/HeaderReader.cs

```diff
@@ -19,17 +21,10 @@ public HeaderReader(IDictionary<string, DataClassification> headersToLog, IRedac
     {
         _redactorProvider = redactorProvider;
 
-        _headers = headersToLog.Count == 0
-            ? Array.Empty<KeyValuePair<string, DataClassification>>()
-            : headersToLog.ToArray();
+        _headers = headersToLog.Count == 0 ? [] : headersToLog.ToArray();
     }
 
-    /// <summary>
-    /// Reads headers and applies filtering (if required).
-    /// </summary>
-    /// <param name="headers">A collection of headers to be read from.</param>
-    /// <param name="listToFill">A list to be filled with headers.</param>
-    public void Read(IHeaderDictionary headers, List<KeyValuePair<string, string>> listToFill)
+    public void Read(IHeaderDictionary headers, IList<KeyValuePair<string, object?>> logContext, string prefix)
```

Comment:
Why make this an IList? That makes the call to Add() below an interface dispatch instead of an inlined call.

Reply:
It's coming directly from AspNetCore, we don't have a choice here. https://github.com/dotnet/aspnetcore/blob/933bc8447e10e3e4a327e8ac1c598935ddd294ae/src/Middleware/HttpLogging/src/HttpLoggingInterceptorContext.cs#L81